python/lsst/ap/association/dia_collection.py

```diff
@@ -19,3 +19,188 @@
 # the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
+
+from __future__ import absolute_import, division, print_function
+
+from scipy.spatial import cKDtree
+
+import .dia_object
+
+
+class DIAObjectCollection(object):
+    """ A collection of DIAObjects with convenience functions for scoring and
+    matching DIASources into the collection of DIAObjects.
+
+    Attributes
+    ----------
+    dia_objects : a list of DIAObjects
+        List of DIAObjects representing this collection the current (e.g.)
```

Comment:
Rather than storing the DIAObjects as a list, I think it may make sense to store them in a dictionary for fast access by id.  I am open to counterarguments, though.